llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -4138,9 +4138,11 @@ void SelectionDAGBuilder::visitAlloca(const AllocaInst &I) {
                                           APInt(IntPtr.getScalarSizeInBits(),
                                                 TySize.getKnownMinValue())));
   else
-    AllocSize =
-        DAG.getNode(ISD::MUL, dl, IntPtr, AllocSize,
-                    DAG.getConstant(TySize.getFixedValue(), dl, IntPtr));
+    AllocSize = DAG.getNode(
+        ISD::MUL, dl, IntPtr, AllocSize,
+        DAG.getZExtOrTrunc(
+            DAG.getConstant(TySize.getFixedValue(), dl, MVT::getIntegerVT(64)),
```

Comment:
Something weird happened with the formatting here, I think a separate variable for the constant would help

Reply:
Thanks! Fixed.